### PR TITLE
Machine merged

### PR DESCRIPTION
**Pull request by scottnath**
@Snugug 

resolves https://github.ibm.com/punchcard/punchcard-content-types/issues/15

### CODE REVIEW

File: lib/content-types.js [hunk 1]
```diff
@@ -18,11 +26,31 @@ const findCT = (type, types) => {
...
+      }
+
+      // add it to an array if it's not already an array
+      if (!Array.isArray(contentType)) {
+        contentType = [contentType];
+      }
+      resolve(merge(contentType));
+    });
+  }
+
```
Comment (Snugug) [inline lib/content-types.js]: * Type checking should move to `merged`
* should be the following:

```javascript
const merged = (content) => {
  if (content) {
    return merged(content);
  }

  return raw().then(types => {
    return merge(types)
  });
}
```

File: lib/content-types.js [hunk 2]
```diff
@@ -18,11 +26,31 @@ const findCT = (type, types) => {
...
+ * Merge content type object with input plugin data
+ *
+ * @param {object|array} ctObj content type object
+ */
+const merged = (ctObj) => {
+  if (ctObj) {
+    return new Promise((resolve, reject) => {
+      let contentType = ctObj;
+      if (typeof contentType !== 'object') {
+        reject(new Error('Content type object must be an object or an array'));
```
Comment (Snugug) [inline lib/content-types.js]: Don't allow object, only allow array

File: lib/content-types/load.js [hunk 3]
```diff
@@ -34,20 +34,23 @@ const load = () => {
...
         reject(new Error('Content types require a name'));
       }
 
-      if (machineNames.indexOf(machine) === -1) {
-        machineNames.push(name);
-        config.machine = machine;
+      if (!config.hasOwnProperty('id')) {
+        reject(new Error('Content types require an id'));
+      }
+      const id = kebab(config.id);
```
Comment (Snugug) [inline lib/content-types/load.js]: You're checking to see if the ID exists, but not checking to see that the ID the user has included equals the kebab'd version of the ID. You're just overriding that. We don't want to override it, we want to reject if it's not ===

```javascript
if (config.id !== kebab(config.id) {
  reject new Error(`${config.id} needs to be written in kebab case (e.g. ${kebab(config.id)}`);
}
````

File: lib/content-types/load.js [hunk 4]
```diff
@@ -34,20 +34,23 @@ const load = () => {
...
+
+      if (ids.indexOf(id) === -1) {
+        ids.push(id);
+        config.id = id;
         types.push(config);
       }
       else {
-        reject(new Error(`Content type ${machine} is duplicated!`));
+        reject(new Error(`Content type ${id} is duplicated!`));
```
Comment (Snugug) [inline lib/content-types/load.js]: `${config.id}`

File: lib/content-types/load.js [hunk 5]
```diff
@@ -73,9 +76,16 @@ const names = types => {
   });
 };
 
-
-const attributes = (machine, types) => {
-  const filtered = util.singleItem('machine', machine, types);
+/**
+ * Attributes of a Content Type
+ *
+ * @param {string} id  kebab id of a content type
```
Comment (Snugug) [inline lib/content-types/load.js]: Not `kebab id`, just `id`

File: tests/fixtures/content-types/bar.yml [hunk 6]
```diff
@@ -1,5 +1,6 @@
 name: Content Type BAR
 description: Bar Baz Foo
+id: content-type-bar
```
Comment (Snugug) [inline tests/fixtures/content-types/bar.yml]: Can we make this just `bar`? This is what's going to be in the URL

File: tests/fixtures/content-types/foo.yml [hunk 7]
```diff
@@ -1,5 +1,6 @@
 name: Content Type FOO
 description: Foo Bar Baz
+id: content-type-foo
```
Comment (Snugug) [inline tests/fixtures/content-types/foo.yml]: Can we make this just `foo`? This is what's going to be in the URL

**Comment by scottnath** [comment]
@Snugug fixes added in

File: README.md [hunk 8]
```diff
@@ -84,11 +84,35 @@ contentTypes().then(types => {
 });
 ```
 
+#### Merged Content Type using config
+
+Will return an `Array` of content type definitions (that contains only one content type), each being an `Object`, with attributes replaced by individual Input Plugin instances that have been merged with the raw content type options. There are no parameters to pass in.
+
+```javascript
+const contentTypes = require('punchcard-content-types');
+const fooContentObj = {
```
Comment (Snugug) [inline README.md]: This needs to be an array of objects, not a stand-alone object

**Comment by Snugug** [comment]
@scottnath one tiny nitpick, then GTG

File: README.md (hunk 8 above)
Comment (scottnath) [inline README.md]: @Snugug fix added

**Comment by Snugug** [comment]
We should follow up as to why code coverage went down, but going to merge

**Pull request merged**